pkg/controller/core/clusterqueue_controller.go

```diff
@@ -481,6 +568,41 @@ func (h *cqNamespaceHandler) Update(ctx context.Context, e event.UpdateEvent, _
 		}
 	}
 	h.qManager.QueueInadmissibleWorkloads(ctx, cqs)
+
+	if !features.Enabled(features.DefaultLocalQueue) {
+		return
+	}
+
+	if h.namespaceSelector != nil && !h.namespaceSelector.Matches(labels.Set(newNs.Labels)) {
+		return
+	}
+
+	log := ctrl.LoggerFrom(ctx)
+	log.V(2).Info("Namespace update event", "namespace", newNs.Name)
+	for cqRef := range newMatchingCqs {
+		if oldMatchingCqs.Has(cqRef) {
+			continue
+		}
```

Comment:
Nit: wouldn't this be equivalent?

```suggestion
	for cqRef := range cqs {
```